packages/dd-trace/test/plugins/outbound.spec.js

```diff
@@ -157,4 +157,75 @@ describe('OuboundPlugin', () => {
       })
     })
   })
+
+  describe('code origin tags', () => {
+    let instance = null
+
+    beforeEach(() => {
+      const tracerStub = {
+        _tracer: {
+          startSpan: sinon.stub().returns({
+            addTags: sinon.spy()
+          })
+        }
+      }
+      instance = new OutboundPlugin(tracerStub)
+    })
+
+    it('should not add exit tags to span if codeOriginForSpansEnabled is false', () => {
+      sinon.stub(instance, '_tracerConfig').value({ codeOriginForSpansEnabled: false })
+      const span = instance.startSpan('test')
+      expect(span.addTags).to.not.have.been.called
+    })
+
+    it('should add exit tags to span if codeOriginForSpansEnabled is true', () => {
+      sinon.stub(instance, '_tracerConfig').value({ codeOriginForSpansEnabled: true })
+
+      const lineNumber = getNextLineNumber()
+      const span = instance.startSpan('test')
+
+      expect(span.addTags).to.have.been.calledOnce
+      const args = span.addTags.args[0]
+      expect(args).to.have.property('length', 1)
+      const tags = parseTags(args[0])
+
+      expect(tags).to.nested.include({ '_dd.code_origin.type': 'exit' })
+      expect(tags._dd.code_origin).to.have.property('frames').to.be.an('array').with.length.above(0)
+
+      for (const frame of tags._dd.code_origin.frames) {
+        expect(frame).to.have.property('file', __filename)
+        expect(frame).to.have.property('line').to.match(/^\d+$/)
+        expect(frame).to.have.property('column').to.match(/^\d+$/)
+        expect(frame).to.have.property('type').to.a('string')
+      }
+
+      const topFrame = tags._dd.code_origin.frames[0]
+      expect(topFrame).to.have.property('line', lineNumber)
+    })
+  })
 })
+
+function getNextLineNumber () {
+  return String(Number(new Error().stack.split('\n')[2].match(/:(\d+):/)[1]) + 1)
+}
+
+function parseTags (tags) {
```

Comment:
Do we already have a utility function that does this that I can re-use? Seem very much overkill to write this code for this test, but it does make the actual test much simpler to write once the tags are converted to a proper object structure 🤔

Reply:
I suspect https://github.com/DataDog/dd-trace-js/blob/master/packages/datadog-core/src/utils/src/set.js might fit the bill.

Reply:
> Do we already have a utility function that does this that I can re-use?

Not sure we have one right now, but it could definitely be added to `datadog-core/src/util` instead of having the code in this file.

Reply:
I've updated the code now and moved the function to `datadog-core/src/util`. Let me know what you think.